src/Medidata.ZipkinTracer.Core/SpanTracer.cs

```diff
@@ -81,11 +81,11 @@ public virtual Span SendClientSpan(string requestName, string traceId, string pa
             return newSpan;
         }
 
-        public virtual void ReceiveClientSpan(Span span, string clientServiceName)
+        public virtual void ReceiveClientSpan(Span span)
         {
             var annotation = new Annotation()
             {
-                Host = zipkinEndpoint.GetEndpoint(clientServiceName),
+                Host = span.Annotations[0].Host,
```

Comment:
@lschreck-mdsol I think you are right. Though usage would most likely not alter span parameter but exception logging must be as detailed as possible for this. Please check my updates here regarding your comment. 82d0ac8

Thank you